gpii/node_modules/flowManager/src/BrowserChannel.js

```diff
@@ -0,0 +1,37 @@
+(function () {
+
+    "use strict";
+
+    var fluid = require("infusion");
+    var gpii = fluid.registerNamespace("gpii");
+
+    fluid.registerNamespace("gpii.request.flowManager");
+
+    fluid.defaults("gpii.flowManager.browserChannel", {
+        gradeNames: ["fluid.standardRelayComponent", "autoInit"]
+    });
+
+    fluid.defaults("kettle.requests.request.handler.browserChannel", {
+        gradeNames: ["gpii.request.flowManager.sessionAware", "autoInit"],
+        invokers: {
+            handle: {
+                funcName: "gpii.request.flowManager.browserChannelHandle",
+                args: ["{request}"],
+                dynamic: true
+            }
+        }
+    });
+
+    gpii.request.flowManager.browserChannelHandle = function (fluidRequest) {
+        var data = fluidRequest.data;
```

Comment:
You should instead fire the fluidRequest's "onError" event, which has listeners both here and in request.js: https://github.com/fluid-project/kettle/blob/master/lib/request.io.js#L50 - the argument that you supply to the event will be sent via socket.emit . 
The existing strategy should also have succeeded in disconnecting the client and ending the request, but this way should be more reliable.